aea/cli/ipfs_hash.py

```diff
@@ -0,0 +1,491 @@
+    :param text: the content of a configuration file.
+    :return: the updated content of the configuration file.
+    """
+
+    def to_row(x: Tuple[str, str]) -> str:
+        return x[0] + ": " + x[1]
+
+    replacement = "\nfingerprint:\n  {}\n".format(
+        "\n  ".join(map(to_row, sorted(fingerprint_dict.items())))
+    )
+    return re.sub(r"\nfingerprint:\W*\n(?:\W+.*\n)*", replacement, text)
+
+
+def compute_fingerprint(  # pylint: disable=unsubscriptable-object
+    package_path: Path, fingerprint_ignore_patterns: Optional[Collection[str]],
+) -> Dict[str, str]:
+    """
+    Compute the fingerprint of a package.
+
+    :param package_path: path to the package.
+    :param fingerprint_ignore_patterns: filename patterns whose matches will be ignored.
+    :return: the fingerprint
+    """
+    fingerprint = _compute_fingerprint(
+        package_path, ignore_patterns=fingerprint_ignore_patterns,
+    )
+    assert_hash_consistency(fingerprint, package_path)
+    return fingerprint
+
+
+def update_fingerprint(configuration: PackageConfiguration) -> None:
+    """
+    Update the fingerprint of a package.
+
+    :param configuration: the configuration object.
+    :return: None
+    """
+    # we don't process agent configurations
+    if isinstance(configuration, AgentConfig):
+        return
+
+    if configuration.directory is None:
+        raise ValueError("configuration.directory cannot be None.")
+
+    fingerprint = compute_fingerprint(
+        configuration.directory, configuration.fingerprint_ignore_patterns
+    )
+    config_filepath = (
+        configuration.directory / configuration.default_configuration_filename
+    )
+    old_content = config_filepath.read_text()
+    new_content = _replace_fingerprint_non_invasive(fingerprint, old_content)
+    config_filepath.write_text(new_content)
+
+
+def check_fingerprint(configuration: PackageConfiguration) -> bool:
+    """
+    Check the fingerprint of a package, given the loaded configuration file.
+
+    :param configuration: the configuration object.
+    :return: True if the fingerprint match, False otherwise.
+    """
+    # we don't process agent configurations
+    if isinstance(configuration, AgentConfig):
+        return True
+
+    if configuration.directory is None:
+        raise ValueError("configuration.directory cannot be None.")
+
+    expected_fingerprint = compute_fingerprint(
+        configuration.directory, configuration.fingerprint_ignore_patterns
+    )
+    actual_fingerprint = configuration.fingerprint
+    result = expected_fingerprint == actual_fingerprint
+    if not result:
+        click.echo(
+            "Fingerprints do not match for {} in {}".format(
+                configuration.name, configuration.directory
+            )
+        )
+    return result
+
+
+def update_hashes(
+    packages_dir: Path,
+    test_data: Optional[Path] = None,
+    root_dir: Optional[Path] = None,
+    no_wrap: bool = False,
```

Comment:
we no have a mix of `wrap` and `no_wrap`. In fact you negate it below when you pass it to `hash_package`